src/cmd/run.js

```diff
@@ -322,6 +322,8 @@ export class ExtensionRunner {
 
   run(profile: FirefoxProfile): Promise<FirefoxProcess> {
     const {firefoxApp, firefox} = this;
-    return firefoxApp.run(profile, {firefoxBinary: firefox});
+    return firefoxApp.run(
+        profile, {firefoxBinary: firefox, binaryArgs: ['-jsconsole']}
```

Comment:
You're on the right track but this code will always run Firefox with `-jsconsole`. We only want to run with that arg when the user specifies it as `web-ext run --jsconsole`. 

Because you've already defined `jsconsole` in `program.js`, the [run()](https://github.com/mozilla/web-ext/blob/master/src/cmd/run.js#L162) function will be invoked like this when someone runs `web-ext run --jsconsole`:

````javascript
run({
  sourceDir: '.', 
  artifactsDir: 'web-ext-artifacts',
  jsconsole: true,
});
````
You need to pass that into `ExtensionRunner.run()` and use it like:
````javascript
const binaryArgs = [];
if (jsconsole) {
  binaryArgs.push('-jsconsole');
}
return firefoxApp.run(
  profile, {firefoxBinary: firefox, binaryArgs}
````